src/MediaCollection/MediaCollection.php

```diff
@@ -20,8 +20,8 @@ class MediaCollection
     /** @var callable */
     public $acceptsFile;
 
-    /** @var bool */
-    public $singleFile = false;
```

Comment:
As this and the Media Collections are public, is this not a potential problem for bc?

Reply:
I don't think so. This variable was only set in `MediaCollection.php` (via the `singleFile()` method) and checked in `FileAdder.php`. If you would upgrade to this version, everything gets converted to these changes under the hood and don't need any additional changes in the public codebase.

Reply:
Apologies, I should have provided an example.

In my models it is possible for me to do the below currently

`if($this->mediaCollections[0]->singleFile) {
            //
        }`

As `$mediaCollections` is public and so is `$singleFile`.

However renaming `$singleFile` to `$collectionSizeLimit` would result in an error in any code which did read the variable directly via it's public access.

Reply:
My bad, misunderstood what you meant. In that case, yes.. that could cause applications to crash. Not sure what is the right way to go, my feeling says:

1. Restore `$singleFile` as public variable to `MediaCollection.php`
2. Whenever `onlyKeepsLatest()` is called set the variable to true if `n=1`

This could later be refactored and removed in a new release. But I leave that up to @freekmurze and the people maintaining this package to decide.

Reply:
I would restore it and flag it as `@deprecated` to hint all IDEs. Another way would be to implement a `__get()` method which returns this value on a dynamic approach - because in theory you could also do `$mediaCollection-> collectionSizeLimit = 5;` because it's public. And having it overloaded would also prevent it from getting deprecated because the `singleFile()` method will also stay there!? So there should be a corresponding getter/property.

Reply:
Was just seeing what it would take to implement something similar to the below on the setter -

```
if($int === 1) {
    $this->singleFile = true;
} else {
    $this->singleFile = false;
}

$this->collectionSizeLimit = $int;
```

But, after mulling it over. Perhaps as @Gummibeer mentioned a `__get()` method would be better; when trying to obtain the `$singleFile` value, `if($collectionSizeLimit === 1) { return true; }` etc.

Unsure as to whether the below would be required as well on the setter? Without looking into it I'm not sure how it would handle a 0 or negative value?

```
public function onlyKeepLatest(int $int): self
{
    if ($int <= 0) {
        // Should an error be thrown??? Should it default to a value???
    }
    ...
}
```

Reply:
For the `$mediaCollection->singleFile` I would go with the `__get` - it's the cleanest solution and handles all circumstances.

The below `1` values are also an interesting special case - if using a default value we should switch to `false` (no limit). But throwing an `OutOfBoundsException` seems better to prevent doing some "magic" if it's just the result of a wrong written math logic. So the user should handle it.

But I would use `0` to remove the limit - so set it to `false`. Because otherwise we won't have a way to remove it again.

```php
if($int === 0) {
  $this->collectionSizeLimit = false;
}

if($int < 0) {
  throw new \OutOfBoundsException();
}

$this->collectionSizeLimit = $int;
```

Reply:
Should we also make the `$collectionSizeLimit` private on `MediaCollection.php`? I see no reason why it should be accessible or overwritable on runtime since that defeats the purpose it has in the first place.

Reply:
Because there are no getters for all the properties they are all `public`. Otherwise I can't use this value in my own logic extending the package.

But making them all private and adding corresponding getter methods would be a BC.

So far I know you also can't overload/read private properties via `__get`!? If this would be possible I would make them `private` and make them available via `__get`. If this sounds like a good idea but using the same name isn't possible we could get dirty and prefix the real properties with `_`.

Reply:
Point taken, I won't change that part. Working on the BC fix for `$singleFile` right now.